util/helm/helm_test.go

```diff
@@ -184,3 +184,23 @@ func TestHelmArgCleaner(t *testing.T) {
 	assert.Contains(t, argsToBeCleaned, `b\,a\,r`)
 
 }
+
+func TestHelmValues(t *testing.T) {
+	h := NewHelmApp("./testdata/redis", []*argoappv1.HelmRepository{})
+	opts := argoappv1.ApplicationSourceHelm{
+		ValueFiles: []string{"values-production.yaml"},
+		Values: `cluster:
+  slaveCount: 2
+`,
```

Comment:
I wonder if we shouldn't test an override of a values.yaml with the provided overrides, to ensure the generated content of values into values.yaml is not messing up with existing values.yaml ?

Reply:
Ofc, I should think twice before reviewing. I see this util/helm/helm.go that it would be no problem, but I still think increasing test coverage could help secure things for the future.